admin/onpage/class-onpage.php

```diff
@@ -0,0 +1,170 @@
+		}
+	}
+
+	/**
+	 * Adding a weekly schedule to the schedules array
+	 *
+	 * @param array $schedules Array with schedules.
+	 *
+	 * @return array
+	 */
+	public function add_weekly_schedule( array $schedules ) {
+		$schedules['weekly'] = array( 'interval' => WEEK_IN_SECONDS, 'display' => __( 'Once Weekly' ) );
+
+		return $schedules;
+	}
+
+	/**
+	 * Fetching the data from onpage.
+	 *
+	 * @return bool
+	 */
+	public function fetch_from_onpage() {
+		if ( $this->onpage_option->can_fetch() ) {
+			// Updates the timestamp in the option.
+			$this->onpage_option->set( 'last_fetch', time() );
+
+			// The currently indexability status.
+			$old_status = $this->onpage_option->get( 'status' );
+
+			// The requested indexability status.
+			$new_status = $this->request_indexability();
+
+			// Saving the new status.
+			$this->onpage_option->set( 'status', $new_status );
+
+			// Check if the status has been changed.
+			if ( $old_status !== $new_status ) {
+				$this->notify_admins( $old_status, $new_status );
+			}
+
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Show a notice when the website is not indexable
+	 */
+	public function show_notice() {
+		$show_notice = WPSEO_Utils::grant_access() && $this->user_has_not_dismissed() && ! $this->onpage_option->is_indexable();
+		if ( $show_notice ) {
+			Yoast_Notification_Center::get()->add_notification(
+				new Yoast_Notification(
+					__( 'Your site is currently not indexable.', 'wordpress-seo' ),
+					array(
+						'type'  => 'error yoast-dismissible',
+						'id'    => 'wpseo-dismiss-onpageorg',
+						'nonce' => wp_create_nonce( 'wpseo-dismiss-onpageorg' ),
+					)
+				)
+			);
+		}
+	}
+
+	/**
+	 * Sending a request to OnPage.org to get the indexability
+	 *
+	 * @return int
+	 */
+	protected function request_indexability() {
+		$request  = new WPSEO_OnPage_Request( home_url() );
+		$response = $request->get_response();
+
+		return (int) $response['is_indexable'];
+	}
+
+	/**
+	 * Notify the admins
+	 *
+	 * @param int|null $old_status The old indexable status.
+	 * @param int      $new_status The new indexable status.
+	 */
+	protected function notify_admins( $old_status, $new_status ) {
+		new WPSEO_OnPage_Notify( $old_status, $new_status );
+	}
+
+	/**
+	 * Setting up the hooks.
+	 */
+	private function set_hooks() {
+		// Schedule cronjob when it doesn't exists on activation.
+		register_activation_hook( WPSEO_FILE, array( $this, 'activate_hooks' ) );
+
+		// Add weekly schedule to the cron job schedules.
+		add_filter( 'cron_schedules', array( $this, 'add_weekly_schedule' ) );
+
+		// Adding admin notice if necessary.
+		add_filter( 'admin_init', array( $this, 'show_notice' ) );
+
+		// Setting the action for the OnPage fetch.
+		add_action( 'wpseo_onpage_fetch', array( $this, 'fetch_from_onpage' ) );
+	}
+
+	/**
+	 * The hooks to run on plugin activation
+	 */
+	private function activate_hooks() {
+		$this->set_cron();
+		$this->fetch_from_onpage();
```

Comment:
I don't think it's a good idea to fetch the status immediately after activating the plugin, if the request timeouts it could take a long time to activate the plugin without it being our fault.

Because we call `wp_schedule_event` with `time()` as the first argument the fetching will already be done ASAP.